backend/WebApi/Services/UserService.cs

```diff
@@ -64,12 +64,23 @@ public async Task<bool> DeleteUser(LfId<User> id)
         return result.DeletedCount > 0;
     }
 
-    public async Task<LfUser> GetUserByEmail(string email)
+    public async Task<LfUser?> FindLfUser(string email)
     {
-        return await _systemDbContext.Users
+        var user = await _systemDbContext.Users
             .Find(user => user.Email == email)
-            .Project(user => new LfUser(user.Email, user.Id, user.Role))
             .SingleOrDefaultAsync();
+
+        if (user == null)
+        {
+            return null;
+        }
+
+        var projectRoles = await _systemDbContext.Projects
+            .Find(proj => user.Projects.Contains(proj.Id) && proj.Users.ContainsKey(user.Id))
```

Comment:
this seems odd, we have to check that user contains project and project contains user. I know that's how the data is structured so we can't change it but I don't like it.

Reply:
Yeah. And of course this isn't perfect, because...What if it's only in one list? Who says we should throw it away?

Reply:
Yeah, I'll add it as a discussion item for our meeting